app/models/paper.rb

```diff
@@ -12,6 +12,7 @@ class Paper < ApplicationRecord
 
   accepts_nested_attributes_for :authors, reject_if: proc { |attributes| attributes['name'].blank? }
   accepts_nested_attributes_for :links
+  validates_associated :links
```

Comment:
so I'm doing all of the updating of these paper attributes through papers#update.  I have the validates associated for the new links so it validates its a URL and checks for uniqueness.